plugins/modules/accesspoint_workflow_manager.py

```diff
@@ -0,0 +1,521 @@
+        except Exception as e:
+            self.log("Unable to Login DNAC "+ str(e) , "ERROR")
+
+    # Below function used pydentic validation over the ansible validation
+    # We can customize validation, if it not required we can remove this function.
+    def validate_input(self, inputdata):
+        """
+        Validate the fields provided in the yml files.
+        Checks the configuration provided in the playbook against a predefined specification
+        to ensure it adheres to the expected structure and data types based pydentics package.
+        Parameters:
+          - inputdata: To validate the input file of yaml keys values will be validated.
+        Returns:
+          The method not returns anything just validation input if anything worng will stop execution.
+        Example:
+            To use this method, create an instance of the class and call 'validate_input' on it.
+          If the validation succeeds, this will allow to go next step, unless this will stop execution.
+          based on the fields.
+        """
+        self.log('Validating the Yaml File..', "INFO")
+        try:
+            CheckUrl(hosturl="https://" + inputdata["dnac_host"])
+            CheckNames(names=inputdata["dnac_username"])
+            CheckNames(names=inputdata["dnac_password"])
+            CheckPort(port=inputdata["dnac_port"])
+            aplist = inputdata.get("config")
+            for eachap in aplist:
+                CheckMACaddress(mac_address=eachap["mac_address"])
+                CheckNames(names=eachap["location"])
+                CheckNames(names=eachap["ap_name"])
+                CheckBrightnessLevel(led_brightness_level=int(eachap["led_brightness_level"]))
+                CheckEnabledDisabledStatus(EnabledDisabledStatus=eachap["led_status"])
+                #CheckRadioType(ap_radiotype=int(eachap["accesspointradiotype"]))
+            self.log("Successfully validated playbook config ", "INFO")
+            self.msg = "Successfully validated input from the playbook"
+            self.status = "success"
+            return self
+        except ValidationError as e:
+            self.log("Invalid Param provided in input Yml File." + str(e), "ERROR")
+            self.msg = "Invalid parameters in playbook: " + str(e)
+            self.status = "failed"
+            return self
+
+    # Below function used to show the current state of the network divices
+    # As Accesss point Device info, Config and Final field to pass in the Update data
+    def get_have(self):
+        """
+        This function used to get AP device details as json response from DNAC site.
+        by giving MAC address as a input in the URL GET Method
+        Device information given in the input file
+        Device Configuration Details 
+        Conpare with input data and current config data
+        """
+        responses = {}
+        ap_data = []
+        final_ap_data = []
+        devices_list = self.get_state()
```

Comment:
Need to fetch the state in main function itself then invoke the respective api.

Reply:
Now moved to main function.